qubes_config/global_config/usb_devices.py

```diff
@@ -437,38 +348,100 @@ def _verify_additional_vm(self, vm):
         return False
 
     def _initialize_data(self):
-        self._errors.clear()
+        self.problem_fatal_box.set_visible(False)
+
+        # guess at the current sys-usb
+        self.rules, self.current_token = \
+            self.policy_manager.get_rules_from_filename(
+                self.policy_filename, self.default_policy)
+
+        if not self.usb_qubes:
+            self.disable_u2f("No USB qubes found. To use U2F Proxy, you have to"
+                             "connect your USB controller to a qube.")
+            return
+
+        usb_qube_candidates = set()
+        for qube in self.usb_qubes:
+            if qube.features.check_with_template(
+                    self.SUPPORTED_SERVICE_FEATURE):
+                usb_qube_candidates.add(qube)
+
+        self.usb_qubes = usb_qube_candidates
+
+        self.usb_qube_model = VMListModeler(
+            combobox=self.usb_qube_combo,
+            qapp=self.qapp,
+            filter_function=lambda vm: vm in self.usb_qubes,
+            style_changes=True
+        )
+
+        if not usb_qube_candidates:
+            self.disable_u2f(
+                "The Qubes U2F Proxy service is not installed in the USB qube. "
+                "If you wish to use this service, install the "
+                "<tt>qubes-u2f</tt> package in the template on which "
+                "the USB qube is based.")
+            return
+
+        policy_candidates = set()
+        for rule in self.rules:
+            try:
+                policy_candidates.add(self.qapp.domains[rule.target])
+            except KeyError:
+                continue
+
+        if not policy_candidates:
+            sys_usb = next(iter(self.usb_qubes))
+        else:
+            # just grab one
+            sys_usb = policy_candidates.pop()
+
+        while sys_usb not in usb_qube_candidates and policy_candidates:
+            try:
+                sys_usb = policy_candidates.pop()
+            except KeyError:
```

Comment:
The loop condition guarantee it won't fail this way here, no? Was it meant to handle different case, like running out of `policy_candidates` before finding any match?